File: chroma-league/src/test/java/com/bonepl/chromaleague/hud/parts/resource/YasuoWindBarTest.java
package com.bonepl.chromaleague.hud.parts.resource;

import com.bonepl.chromaleague.IntSteps;
import com.bonepl.chromaleague.hud.AnimationTester;
import org.junit.jupiter.api.Test;

class YasuoWindBarTest {
    @Test
    void testYasuoWindBar() {
        final IntSteps intSteps = new IntSteps(0, 100, 5);
        final YasuoWindBar yasuoWindBar = new YasuoWindBar();

        new AnimationTester()
                .withBeforeIterationAction(i -> {
                    if (i >= 20 && i <= 50) {
                        ManaBarTest.mockResource(100, 100);
                    } else {
                        ManaBarTest.mockResource(intSteps.nextInt(), 100);
                    }
                })
                .testAnimation(yasuoWindBar, 70);

    }
}

File: test/hud/parts/resource/YasuoWindBar.test.ts
import {describe, test} from 'vitest';
import {IntSteps} from '../../../IntSteps.js';
import {AnimationTester} from '../../../AnimationTester.js';
import {YasuoWindBar} from '../../../../src/hud/parts/resource/YasuoWindBar.js';
import {mockResource, setupResourceTest} from './AbstractResourceTest.js';

/** Integration test for Yasuo wind resource bar on actual hardware. */
describe('YasuoWind Integration', () => {
    setupResourceTest();

    test('testYasuoWindBar', async () => {
        const intSteps = new IntSteps(0, 100, 5);
        const yasuoWindBar = new YasuoWindBar();

        await new AnimationTester()
            .withBeforeIterationAction(i => {
                if (i >= 20 && i <= 50) {
                    mockResource(100, 100);
                } else {
                    mockResource(intSteps.nextInt(), 100);
                }
            })
            .testAnimation(yasuoWindBar, 70);
    }, 30000);
});
